sdk/nodejs/eventhub/zMixins.ts

```diff
@@ -498,3 +501,180 @@ export class EventHubFunction extends appservice.Function<EventHubContext, strin
+        };
+
+        const { bindings, appSettings } =
+            appservice.combineBindingSettings(trigger, args.inputs, args.outputs);
+
+        super(name, bindings, args, appSettings);
+    }
+}
+
+export interface EventGridCallbackSubscriptionArgs<T> extends appservice.InputOutputsArgs,
+                                                              appservice.CallbackFunctionAppArgs<EventGridContext<T>, EventGridEvent<T>, appservice.FunctionDefaultResponse> {
+    /**
+     * The name of the resource group in which to create the event subscription. [resourceGroup] takes precedence
+     * over [resourceGroupName]. If none of the two is supplied, the Queue's resource group will be used.
+     */
+    resourceGroupName?: pulumi.Input<string>;
+
+    /**
+     * A list of applicable event types that need to be part of the event subscription.
+     */
+    readonly includedEventTypes?: pulumi.Input<pulumi.Input<string>[]>;
+
+    /**
+     * A retry policy block as defined below.
+     */
+    readonly retryPolicy?: pulumi.Input<{ eventTimeToLive: pulumi.Input<number>, maxDeliveryAttempts: pulumi.Input<number> }>;
+
+    /**
+     * A subject filter block as defined below.
+     */
+    readonly subjectFilter?: pulumi.Input<{ caseSensitive?: pulumi.Input<boolean>, subjectBeginsWith?: pulumi.Input<string>, subjectEndsWith?: pulumi.Input<string> }>;
+}
+
+/**
+ * Resource properties to scope an Event Grid subscription to. The shape fits most Azure resources,
+ * so they can be passed directly.
+ */
+export interface EventGridScope {
+    /**
+     * Resource group name to create subscription at, if another resource group is not explicitly
+     * passed in subscription arguments.
+     */
+    resourceGroupName: pulumi.Input<string>;
+
+    /**
+     * Azure Resource ID.
+     */
+    id: pulumi.Input<string>;
+}
+
+/**
+ * A callback-based subscription to events coming from Event Grid. Creates an Azure Function and
+ * an Event Grid Event Subscription with the webhook URL pointing to the Azure Function.
+ */
+export class EventGridCallbackSubscription<T> extends appservice.EventSubscription<EventGridContext<T>, EventGridEvent<T>, appservice.FunctionDefaultResponse> {
+    public readonly subscription: EventGridEventSubscription;
+
+    constructor(name: string,
+                scope: EventGridScope,
+                args: EventGridCallbackSubscriptionArgs<T>,
+                opts: pulumi.ComponentResourceOptions = {}) {
+
+        const { resourceGroupName, location } =
+            appservice.getResourceGroupNameAndLocation(args, pulumi.output(scope.resourceGroupName));
+
+        super("azure:eventhub:EventGridCallbackSubscription",
+              name,
+              new EventGridFunction(name, args),
+              { ...args, resourceGroupName, location },
+              opts);
+
+        const keys = pulumi.output(this.functionApp.getHostKeys());
+        const key = keys.apply(keys => keys.systemKeys["eventgrid_extension"]);

```

Comment:
should be able to write this as: `keys.systemKeys["eventgrid_extension"]` :)

Reply:
How does that work? I thought I can't dot into an `Output`?

Reply:
Yup, we added this back in 0.17: https://www.pulumi.com/blog/simplified-outputs-in-pulumi-0.17.0/